plugins/feeds/public/cisa_kev.py

```diff
@@ -0,0 +1,153 @@
+import logging
+from datetime import datetime, timedelta
+from typing import ClassVar
+
+from core import taskmanager
+from core.schemas import entity, task
+
+
+def _cves_as_dict(data):
+    cves = dict()
+    for vulnerability in data.get('vulnerabilities', []):
+        cve_id = vulnerability.get('cve', {}).get('id', '')
+        if not len(cve_id):
+            continue
+        cves[cve_id] = vulnerability
+    return cves
+
+def _extract_cvss_metric(cve):
+    metrics = cve.get('metrics', {})
+    metric_version = 0
+    for metric in metrics:
+        version = float('.'.join(list(metric.replace('cvssMetricV', ''))))
+        if version > metric_version:
+            metric_version = version
+    if metric_version == 0:
+        return 0, {}
+    metric_str = str(metric_version).replace('.', '')
+    cvss_metric = metrics.get(f'cvssMetricV{metric_str}', [])
+    if len(cvss_metric):
+        return metric_version, cvss_metric[0]
+    else:
+        return 0, {}
+
+class CisaKEV(task.FeedTask):
+    _defaults = {
+        "frequency": timedelta(hours=1),
+        "name": "CisaKEV",
+        "description": "Imports the list of of known exploited vulnerablities published by CISA",
+        "source": "https://www.cisa.gov/known-exploited-vulnerabilities-catalog",
+    }
+
+    CISA_SOURCE: ClassVar["str"] = "https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json"
+    NVD_SOURCE: ClassVar["str"] = "https://services.nvd.nist.gov/rest/json/cves/2.0?hasKev"
+
+    def run(self):
+        response = self._make_request(self.CISA_SOURCE)
+        if not response:
+            logging.warning(f"Unable to fetch feed from {self.CISA_SOURCE}")
+            return
+        kev_json = response.json()
+        response = self._make_request(self.NVD_SOURCE)
+        if not response:
+            logging.warning(f"Unable to fetch feed from {self.NVD_SOURCE}")
+            nvd_json = {}
+        else:
+            nvd_json = _cves_as_dict(response.json())
+        for entry in kev_json.get('vulnerabilities', list()):
+            cve_id = entry.get('cveID', '')
+            if not len(cve_id):
+                continue
+            cve_details = nvd_json.get(cve_id, {})
+            self.analyze_entry(entry, cve_details)
+
+
+    def _analyze_cve_details(self, cve_details: dict):
+        """Analyzes an entry as specified in nist nvd json."""
+        if not isinstance(cve_details, dict) or len(cve_details) == 0:
+            return 0, 'none', ''
+        cve = cve_details.get('cve')
```

Comment:
```suggestion
        cve = cve_details['cve']
```

(or `.get('cve', {})` if it's likely to be missing)

The whole logic under this (e.g. L76) assumes cve is a dict. (

Reply:
I'm more in favor of using .get('cve', {})